docs/src/howto-guides/029_mqtt_local_broker_authentication.md

```diff
@@ -0,0 +1,166 @@
+  authenticate itself to connecting clients
+- `keyfile`: points to a private key of the specified certificate, necessary for
+  encrypted communication
+
+Be sure that mosquitto can read both files, especially the private key, which
+can only be read by the owner. Set `mosquitto:mosquitto` as an owner of these
+files. If you're unsure where to place them, `/etc/mosquitto/ca_certificates`
+is the directory intended for them, although you can use other paths if
+necessary.
+
+The certificates used need to be X.509 v3 certificates with a `subjectAltName`
+section containing the hostname that the broker is running on.
+
+### Step 2: Configure thin-edge.io to connect to the new listener
+
+Execute the following commands:
+
+```sh
+tedge config set mqtt.client.port 8883
+tedge config set mqtt.client.ca_file PATH_TO_CA_CERTIFICATE
+# optional
+tedge config set mqtt.client.ca_dir PATH_TO_CA_CERTIFICATE_DIRECTORY
+```
+
+`mqtt.client.ca_file` and `mqtt.client.ca_dir` options point to trusted CA
+certificate(s) used to verify the broker. If either is used, server
+authentication is enabled.
+
+### Step 3: Restart services
+
+Now you will need to manually restart all the affected services so that they can
+pick up the configuration change.
+
+## Server + client authentication
+
+Additionally, the server can require connecting clients to present a valid
+certificate. These client certificates need to be signed by a CA that the server
+trusts. CA used to sign a server certificate and CA signing client certificates
+do not have to be the same.
+
+### Step 1: Configure server + client authenticated listener in Mosquitto broker
+
+Change the content of the conf file defined previously to the following and
+restart mosquitto service:
+
+```conf
+listener 8883
+allow_anonymous false
+require_certificate true
+cafile   PATH_TO_SERVER_CERTIFICATE
+certfile PATH_TO_SERVER_PRIVATE_KEY
+keyfile  PATH_TO_CLIENT_CA_CERTIFICATE
+```
+
+- `allow_anonymous` disables anonymous access to the listener; connecting
+  clients will need to authenticate themselves
+- `require_certificate` requires clients to provide a certificate as a means of
+  authentication
+
+### Step 2: Configure thin-edge.io to use a client certificate and private key
+
+```sh
+tedge config set mqtt.client.auth.cert_file PATH_TO_CLIENT_CERTIFICATE
+tedge config set mqtt.client.auth.key_file PATH_TO_CLIENT_PRIVATE_KEY
+```
+
+Both `cert_file` and `key_file` are required to enable client authentication.
+Setting only one of them will result in an error about the second one not being
+set.
+
+As with the server private key, set `tedge:tedge` as the owner of the
+certificate and the private key, so that the private key can be read by
+thin-edge components.
+
+### Step 3: Restart services
+
+Now you will need to manually restart all the affected services so that they can
+pick up the configuration change.
+
+## Generating certificates
+
+You can use the following script to generate all required certificates:
```

Comment:
@reubenmiller We can consider to extend the `tedge cert create` command to create such certificates.